src/generate/mod.rs

```diff
@@ -59,26 +60,7 @@ pub type DesugarInput = (AST, Option<String>, Option<PathBuf>);
 ///
 /// A malformed [AST](crate::parser::ast::AST) causes this stage
 /// to panic.
-pub fn gen(input: &AST) -> GenResult {
-    convert_node(input, &mut Imports::new(), &State::new())
-}
-
-pub fn gen_all(inputs: &[DesugarInput]) -> GenResults {
-    let inputs: Vec<_> = inputs
-        .iter()
-        .map(|(ast, source, path)| (gen(ast), source, path))
-        .map(|(result, source, path)| {
-            (result.map_err(|err| err.into_with_source(source, path)), source.clone(), path.clone())
-        })
-        .collect();
-
-    let (oks, errs): (Vec<_>, Vec<_>) = inputs.iter().partition(|(res, ..)| res.is_ok());
-    if errs.is_empty() {
-        Ok(oks
-            .iter()
-            .map(|(res, src, path)| (res.as_ref().unwrap().clone(), src.clone(), path.clone()))
-            .collect())
-    } else {
-        Err(errs.iter().map(|(res, ..)| res.as_ref().unwrap_err().clone()).collect())
-    }
+pub fn gen(input: &ASTTy) -> GenResult {
+    let ast = AST::new(&input.pos, input.node.clone());
+    convert_node(&ast, &mut Imports::new(), &State::new())
```

Comment:
Change signature of `convert_node` and all inner nodes to make use of `ASTTy`.
This is not a big change, and allows us to also use the type in annotations